python/mpt/src/mpt/ethereum_tries.py

```diff
@@ -352,19 +353,92 @@ def from_data(cls, data: Dict[str, Any]) -> "EthereumTrieTransitionDB":
         instance.post_state_root = post_state_root
         return instance
 
-    def to_pre_state(self) -> State:
-        """Convert the pre-state trie to a State object."""
-        return self.to_state()
 
-    def to_post_state(self) -> State:
-        """Convert the post-state trie to a State object."""
-        state = State()
-        if self.post_state_root not in self.nodes:
-            raise ValueError(
-                f"Post state root not found in nodes: {self.post_state_root}"
+class PreState:
+    @staticmethod
+    def from_data(data: Dict[str, Any]) -> State:
+        """
+        Create a PreState object from the ZKPI-provided data.
+        """
+        pre_state = State()
+        for address_hex, account in data["extra"]["preState"].items():
+            address = Address.fromhex(address_hex[2:])
+            # Create an empty, non-defaultdict, storage trie for the account
+            storage_trie = Trie(secured=True, default=U256(0), _data={})
+            pre_state._storage_tries[address] = storage_trie
+            if account is None:
+                # If the account is not present in the preState data, we want it explicitly set to NONE instead of being a non-existent entry.
+                # This is very important for the args_gen purpose, as we need all touched accounts to be present in the initial state dict.
+                pre_state._main_trie._data[address] = None
+                continue
+
+            # Initialize the account
+            pre_balance = U256(int(account["balance"][2:], 16))
+            pre_nonce = Uint(int(account["nonce"][2:], 16))
+            pre_code_hash = Hash32.fromhex(account["codeHash"][2:])
+            pre_storage_hash = Hash32.fromhex(account["storageHash"][2:])
+            pre_code = Bytes.fromhex(account["code"][2:]) if "code" in account else None
+
+            # Note: If you want to use EELS with the ZKPi-pre-state, you need to explicitly set the code to an empty bytearray if the codehash is none,
```

Comment:
why are we code to None and not bytearray()?

Reply:
None would be a non-cached entry, whereas bytearray() is explicitly saying "this account has for code b''"